infra/build/developer-tools/build/scripts/task_helper_functions.sh

```diff
@@ -384,7 +384,22 @@ function check_headers() {
   echo "Checking file headers"
   # Use the exclusion behavior of find_files
   find_files . -type f -print0 \
-    | compat_xargs -0 python /usr/local/verify_boilerplate/verify_boilerplate.py
+    | compat_xargs -0 python /usr/local/verify_boilerplate/verify_boilerplate.py 2>&1 \
+    | grep -v 'have incorrect boilerplate headers' > legacy_check.log
+  if [[ -v EXCLUDE_HEADER_CHECK ]]; then
+    SKIP_STRING=""
+    for file in ${EXCLUDE_HEADER_CHECK}
+    do
+      SKIP_STRING="$SKIP_STRING -skip $file"
+    done
+    stored_cmd="addlicense -check $SKIP_STRING ./ 2>&1 | grep -v 'skipping this file'"
+    eval "$stored_cmd" > new_check.log
+  else
+    addlicense -check ./ 2>&1 > new_check.log
```

Comment:
We should not check globally. Instead, we should use `find_files` to find the directories to check (like the legacy check does) and then run the check against each.

Reply:
This has been fixed. I have used the `find_files` utility function to get the list of files to check